lib/elixir/test/elixir/fixtures/dialyzer/boolean_check.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Dialyzer.BooleanCheck do
+  def and_check(arg) do
+    arg and arg
+  end
+
+  def and_check_optimized(arg) do
+    is_integer(arg) and arg > 10
+  end
+
+  def or_check(arg) do
+    arg or arg
```

Comment:
The test would be checking our use of generated - which the current file does not do.